packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/components/Regenerate/index.js

```diff
@@ -0,0 +1,86 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+import { Button } from '@strapi/design-system/Button';
+import Refresh from '@strapi/icons/Refresh';
+import { ConfirmDialog } from '@strapi/helper-plugin';
+import { axiosInstance } from '../../../../../../../core/utils';
+
+const ButtonWithRightMargin = styled(Button)`
+  margin-right: ${({ theme }) => theme.spaces[2]};
+`;
+
+export const Regenerate = ({ onRegenerate, idToRegenerate }) => {
+  let isLoadingConfirmation = false;
+  const { formatMessage } = useIntl();
+  const [showConfirmDialog, setShowConfirmDialog] = useState(false);
+  const handleConfirmRegeneration = async () => {
+    isLoadingConfirmation = true;
+    try {
+      const {
+        data: {
+          data: { accessKey },
+        },
+      } = await axiosInstance.post(`/admin/api-tokens/${idToRegenerate}/regenerate`);
+      isLoadingConfirmation = false;
+      onRegenerate(accessKey);
+      setShowConfirmDialog(false);
+    } catch (error) {
+      isLoadingConfirmation = false;
+      console.log('error', error);
```

Comment:
just a note: we would want to remove this log

Reply:
ok there is a way to log the error in any way?

Reply:
yes, something like this, you can find examples of how to import it and use it in the main EditView entry file 
```js
toggleNotification({
          type: 'warning',
          message: { id: 'notification.error', defaultMessage: 'An error occured' },
        });
```

Reply:
ah ok great, thanks

Reply:
done